pkg/planner/core/preprocess.go

```diff
@@ -1634,6 +1634,10 @@ func (p *preprocessor) handleTableName(tn *ast.TableName) {
 			return
 		}
 	}
+	if err := p.checkTableVisible(tableInfo); err != nil {
```

Comment:
I am not sure about this modification, please ask the DDL owner to take a look.

REST LGTM.